### PR TITLE
Support DocC Xcode 13

### PR DESCRIPTION
**Pull request by wendyliga**
## Preview

Screen Shot 2021-06-11 at 19 57 50

## Description

This year, apple release [DocC with Xcode 13](https://developer.apple.com/videos/play/wwdc2021-10166). [DocC](https://developer.apple.com/documentation/docc) is new framework to generate documentation right inside Xcode.The generated documentation can be viewed from `Help > Developer Documentation(⌘⇧0)` or as webpage.

### Add `symbol link` to relevant type on TCA
Screen Shot 2021-06-11 at 19 58 44

`symbol link` is a way to link certain type with wrapping it inside double backtick. on the generated document, it will become a link, and when pressed will redirect to respective page.

### Wrap code snippet inside triple backtick swift, so the code style could be applied on doc preview
|without wrapper| with wrapper|
|---|---|
|Screen Shot 2021-06-11 at 20 00 07|Screen Shot 2021-06-11 at 19 59 41|

_edit: fix grammer_

### CODE REVIEW

**Review by mbrandonw — COMMENTED** [review]
Hey @wendyliga, thanks for this!

We're really excited to try out DocC, but we're not entirely sure how it's going to play with SwiftDoc, which we will need to use for at least a few more months until Xcode 13 is officially released.

What we're going to do is try merging this soon, see how SwiftDoc behaves, and if there are any problems we may revert it and try fixing things.

So, we'll merge soon, but also we have a question for you down below.

File: Sources/ComposableArchitecture/Effect.swift [hunk 1]
```diff
@@ -195,7 +207,7 @@ public struct Effect<Output, Failure: Error>: Publisher {
   ///
   /// - Warning: Combine's `Publishers.Concatenate` operator, which this function uses, can leak
   ///   when its suffix is a `Publishers.MergeMany` operator, which is used throughout the
-  ///   Composable Architecture in functions like `Reducer.combine`.
+  ///   Composable Architecture in functions like `Reducer/combine(_:)-1ern2``.
```
[the remark this answers is not in the document]
Comment (wendyliga) [reply Sources/ComposableArchitecture/Effect.swift:210]: @mbrandonw this is how DocC refer to overload function.
i think they have some hash function, to generate this suffix, from given parameter signature.

[the remark this answers is not in the document]
Comment (mbrandonw) [reply Sources/ComposableArchitecture/Effect.swift:210]: Ah ok, so did Xcode insert this automatically during the compilation process, or did you have to insert it manually?

[the remark this answers is not in the document]
Comment (wendyliga) [reply Sources/ComposableArchitecture/Effect.swift:210]: @mbrandonw automatically, i only refer it from autocomplete

**Comment by wendyliga** [comment]
let me try generate `SwiftDoc` from this branch, to see if this changes can works with `SwiftDoc`

**Comment by wendyliga** [comment]
@mbrandonw i think it will work
Screen Shot 2021-06-14 at 23 12 04

**Comment by mbrandonw** [comment]
> @mbrandonw i think it will work

Ok cool! We'll merge and try this out once tests pass.

Thanks again for taking this on!

**Pull request merged**